WordPress/openverse-frontend
# Make VLink component that wraps around both external and internal links

## Fixes


Fixes #468 by @obulat

## Description


This PR supercedes #468, based on @krysal 's [comment about wrapping internal links, too](https://github.com/WordPress/openverse-frontend/pull/871#pullrequestreview-883719330).

Now, `VLink` is a wrapper component that should be used for all links. It determines whether the link is internal or external by checking if the `href` attr starts with `/`.

**External links:**
- use `a` element.
- add `target='_blank'` that allows the links to open in another tab, instead of causing errors in an iframe.
- adds `rel='noopener noreferrer' to prevent some referrer vulnerabilities in older browsers (`target='_blank'` is enough for modern browsers)

**Internal links:**
- use `NuxtLink` component.
- wrap the `href` in `app.localePath` to ensure that the correct locale is used, and pass it as the `to` prop to `NuxtLink`.

This PR moves the warning for missing `href` from `VButton` to `VLink`, so `VButton` doesn't have to handle the lack of `href` property.

## Testing Instructions

Run the app and test out internal links and external links. Also check that the links in the navbar (pages) work as expected. External links should always open in a new tab.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
